resolwe/flow/models/secret.py

```diff
@@ -0,0 +1,57 @@
+"""Resolwe secrets model."""
+import uuid
+
+from fernet_fields import EncryptedTextField
+
+from django.conf import settings
+from django.db import models
+
+
+class SecretManager(models.Manager):
+    """Manager for Secret objects."""
+
+    def create_secret(self, value, contributor):
+        """Create a new secret, returning its handle.
+
+        :param value: Secret value to store
+        :param contributor: User owning the secret
+        :return: Secret handle
+        """
+        secret = self.create(value=value, contributor=contributor)
+        return str(secret.handle)
+
+    def get_secret(self, handle, contributor=None):
+        """Retrieve an existing secret's value.
+
+        :param handle: Secret handle
+        :param contributor: Optional contributor if the caller wants to
+            perform contributor validation (e.g. only secrets for the
+            given contributor will be looked up).
+        """
+        queryset = self.all()
+        if contributor is not None:
+            queryset = queryset.filter(contributor=contributor)
+        secret = queryset.get(handle=handle)
+        return secret.value
+
+
+class Secret(models.Model):
```

Comment:
I think we will need an extra to describe where the secret comes from. E.g. if you get users API token first time, you don't have to ask for it in following requests.

Reply:
We can add an arbitrary JSON metadata column for that?

Reply:
Done.